test/plugins/retry429.js

```diff
@@ -0,0 +1,558 @@
+// Copyright © 2017 IBM Corp. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* global describe it before after */
+'use strict';
+
+const assert = require('assert');
+const Client = require('../../plugins/retry.js');
+const nock = require('../nock.js');
+
+const ME = process.env.cloudant_username || 'nodejs';
+const PASSWORD = process.env.cloudant_password || 'sjedon';
+const SERVER = 'https://' + ME + '.cloudant.com';
+
+describe('Retry429 Plugin', function() {
+  before(function(done) {
+    var mocks = nock(SERVER)
+        .put('/foo')
+        .reply(201, {ok: true});
+
+    var cloudantClient = new Client({ https: true });
+
+    var req = {
+      url: SERVER + '/foo',
+      auth: { username: ME, password: PASSWORD },
+      method: 'PUT'
+    };
+    cloudantClient(req, function(err, resp) {
+      assert.equal(err, null);
+      assert.equal(resp.statusCode, 201);
+      mocks.done();
+      done();
+    });
+  });
+
+  after(function(done) {
+    var mocks = nock(SERVER)
+        .delete('/foo')
+        .reply(200, {ok: true});
+
+    var cloudantClient = new Client({ https: true });
+
+    var req = {
+      url: SERVER + '/foo',
+      auth: { username: ME, password: PASSWORD },
+      method: 'DELETE'
+    };
+    cloudantClient(req, function(err, resp) {
+      assert.equal(err, null);
+      assert.equal(resp.statusCode, 200);
+      mocks.done();
+      done();
+    });
+  });
+
+  describe('with callback only', function() {
+    it('performs request and returns response', function(done) {
```

Comment:
I think this test could use a comment about why it runs even with `NOCK_OFF` when the others don't. Somethin to the effect of, "tests only a single request with a good response so can run against a real server" or whatever.